packages/http-caching-proxy/src/http-caching-proxy.ts

```diff
@@ -124,7 +132,8 @@ export class HttpCachingProxy {
   private _handle(request: IncomingMessage, response: ServerResponse) {
     const onerror = (error: Error) => {
       this.logError(request, error);
-      response.statusCode = error.name === 'RequestError' ? 502 : 500;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      response.statusCode = (error as any).statusCode || 502;
       response.end(error.message);
```

Comment:
Should we preserve the error name (`Error`, `TypeError`, etc.) in the error response body returned by our proxy?

```suggestion
      response.end(error.name + ': ' + error.message);
```